packages/blast-core/lib/bre.test.js

```diff
@@ -0,0 +1,424 @@
+    await mockedSignerInstantiate(signerAddress, codeId, msg, label, instantiateFee, { funds: funds }),
+  execute: async (signerAddress, contractAddress, msg, fee) =>
+    await mockedSignerExecute(signerAddress, contractAddress, msg, fee),
+  queryContractSmart: async (contractAddress, msg) => await mockedSignerQueryContractSmart(contractAddress, msg),
+  address: validSignerAddress
+}
+
+const mockedSigner2 = {
+  mockedData: 'mocked signer data',
+  upload: async (signerAddress, wasmFile, uploadFee) => await mockedSignerUpload(signerAddress, wasmFile, uploadFee),
+  instantiate: async (signerAddress, codeId, msg, label, instantiateFee, { funds }) =>
+    await mockedSignerInstantiate(signerAddress, codeId, msg, label, instantiateFee, { funds: funds }),
+  execute: async (signerAddress, contractAddress, msg, fee) =>
+    await mockedSignerExecute(signerAddress, contractAddress, msg, fee),
+  queryContractSmart: async (contractAddress, msg) => await mockedSignerQueryContractSmart(contractAddress, msg),
+  address: 'cudos1xgejtl9heykzscska6cetsferpelygx8g9vxp9'
+}
+
+const mockedSigners = [mockedSigner1, mockedSigner2]
+
+const mockedSignerUpload = async (signerAddress, wasmFile, uploadFee) => {
+  if (signerAddress === validSignerAddress && wasmFile === 'mockedWasmPathBuffer' && uploadFee) {
+    return mockedUploadTx
+  }
+  throw new Error("Mocked error: A parameter in mocked signer's upload() function is not valid")
+}
+const mockedSignerInstantiate = async (signerAddress, codeId, msg, label, instantiateFee, { funds }) => {
+  if (signerAddress === validSignerAddress && typeof codeId === 'number' && codeId && typeof (msg) === 'object' &&
+    typeof (label) === 'string' && label && instantiateFee) {
+    return mockedInstantiateTx
+  }
+  throw new Error("Mocked error: A parameter in mocked signer's instantiate() function is not valid")
+}
+const mockedSignerExecute = async (signerAddress, contractAddress, msg, fee) => {
+  if (signerAddress === validSignerAddress && contractAddress === validContractAddress &&
+    typeof (msg) === 'object' && fee) {
+    return mockedExecuteTx
+  }
+  throw new Error("Mocked error: A parameter in mocked signer's execute() function is not valid")
+}
+const mockedSignerQueryContractSmart = async (contractAddress, msg) => {
+  if (contractAddress === validContractAddress && typeof (msg) === 'object') {
+    return mockedQueryTx
+  }
+  throw new Error("Mocked error: A parameter in mocked signer's query() function is not valid")
+}
+
+const mockedUploadTx = {
+  originalSize: 128131,
+  originalChecksum: '4ba98b0642afe91d234e32b1f6cb696842977a9cd0a7d8e8e03d5aa85f540dbf',
+  compressedSize: 47480,
+  compressedChecksum: '87b2d334c13486f88517cf8e0db5aa41b181fe64cdba3ef382a6a1c657a65ceb',
+  codeId: 11,
+  logs: [],
+  height: 16074,
+  transactionHash: 'CAB3B37A927C79C181E406926F898CDBFC426CE36918BD6AB6AFA65D4CAEBFFB',
+  gasWanted: 1186959,
+  gasUsed: 924549
+}
+const mockedInstantiateTx = {
```

Comment:
Remove all mocked data from the test file, and create a separate one with only the mocked data. Leave the test file clean with only tests for better review